### PR TITLE
General code optimizations.

### PR DESCRIPTION
**Pull request by brunao05**

### CODE REVIEW

**Comment by rasmus** [comment]
There's a lot of `System.ObjectDisposedException` exceptions thrown in the failing tests, it might be the same root cause.

**Comment by brunao05** [comment]
I have made a new pull, removing unnecessary changes. Please review.

File: Source/EventFlow.Autofac/Registrations/AutofacServiceRegistration.cs [hunk 1]
```diff
@@ -193,12 +201,11 @@ public void Decorate<TService>(Func<IResolverContext, TService, TService> factor
         public IRootResolver CreateResolver(bool validateRegistrations)
         {
             var container = _containerBuilder.Build();
-            var autofacRootResolver = new AutofacRootResolver(container);
-
-            if (validateRegistrations)
-            {
-                autofacRootResolver.ValidateRegistrations();
-            }
+            using (var autofacRootResolver = new AutofacRootResolver(container))
```
Comment (TrayanZapryanov) [inline Source/EventFlow.Autofac/Registrations/AutofacServiceRegistration.cs]: Is it possible disposed exceptions to be because of this using?

**Comment by rasmus** [comment]
Still failing tests

File: Source/EventFlow.Autofac/Registrations/AutofacServiceRegistration.cs (hunk 1 above)
Comment (FinalFortune) [reply Source/EventFlow.Autofac/Registrations/AutofacServiceRegistration.cs]: Assume the container gets disposed with it,  not sure why there's a `return new AutofacRootResolver(container);`. 
 Should remove the `using `and have `return autofacRootResolver`

**Comment by rasmus** [comment]
No CLA signed and many of these are already done, closing.

**Pull request closed without merging**